### PR TITLE
feat: add query_disk_info api for shell command

### PR DESCRIPTION
**Pull request by Shuo-Jia**
## Related issue:
 https://github.com/XiaoMi/pegasus/pull/498, https://github.com/XiaoMi/rdsn/pull/409
## Important: 
It should not be merged until https://github.com/XiaoMi/rdsn/pull/409 is merged

### CODE REVIEW

File: include/dsn/dist/replication/replication_ddl_client.h [hunk 1]
```diff
@@ -222,9 +225,50 @@ class replication_ddl_client
         return error_with<TResponse>(std::move(rpc.response()));
     }
 
+    /// Send request to multi replica server synchronously.
+    template <typename TRpcHolder, typename TResponse = typename TRpcHolder::response_type>
+    std::map<dsn::rpc_address, error_with<TResponse>>
+    call_rpc_async(std::map<dsn::rpc_address, TRpcHolder> &rpcs,
```
Comment (levy5307) [inline include/dsn/dist/replication/replication_ddl_client.h]: `call_rpcs_async`

File: include/dsn/dist/replication/replication_ddl_client.h [hunk 2]
```diff
@@ -222,9 +225,50 @@ class replication_ddl_client
...
+                    } else {
+                        resps.emplace(
+                            rpc.first,
+                            std::move(error_s::make(err, "unable to send rpc to server")));
+                    }
+                });
+        }
+        tracker.wait_outstanding_tasks();
+
+        if (!enable_retry) {
```
Comment (levy5307) [inline include/dsn/dist/replication/replication_ddl_client.h]:
```
        if (enable_retry && rpcs.size() > 0) { 
                std::map<dsn::rpc_address, dsn::error_with<TResponse>> retry_resps = call_rpc_async(rpcs, reply_thread_hash, false);
            for (auto &resp : retry_resps) {
                resps.emplace(resp.first, std::move(resp.second));
            }       
        return resps;
```

File: src/dist/replication/ddl_lib/replication_ddl_client.cpp [hunk 3]
```diff
@@ -1517,5 +1517,20 @@ replication_ddl_client::ddd_diagnose(gpid pid, std::vector<ddd_partition_info> &
 
     return dsn::ERR_OK;
 }
+
+std::map<dsn::rpc_address, error_with<query_disk_info_response>>
+replication_ddl_client::query_disk_info(const std::vector<dsn::rpc_address> &targets, int app_id)
```
Comment (levy5307) [inline src/dist/replication/ddl_lib/replication_ddl_client.cpp]: return map is low efficiency.

File: include/dsn/dist/replication/replication_ddl_client.h (hunk 1 above)
Comment (levy5307) [inline include/dsn/dist/replication/replication_ddl_client.h]: same with ```query_disk_info```

File: include/dsn/dist/replication/replication_ddl_client.h [hunk 4]
```diff
@@ -169,6 +169,11 @@ class replication_ddl_client
 
     dsn::error_code ddd_diagnose(gpid pid, std::vector<ddd_partition_info> &ddd_partitions);
 
+    void
+    query_disk_info(const std::vector<dsn::rpc_address> &targets,
+                    /*out*/ std::map<dsn::rpc_address, error_with<query_disk_info_response>> &resps,
+                    std::string app_name);
```
Comment (acelyc111) [inline include/dsn/dist/replication/replication_ddl_client.h]:
```suggestion
                    const std::string &app_name);
```

File: include/dsn/dist/replication/replication_ddl_client.h [hunk 5]
```diff
@@ -169,6 +169,11 @@ class replication_ddl_client
 
     dsn::error_code ddd_diagnose(gpid pid, std::vector<ddd_partition_info> &ddd_partitions);
 
+    void
+    query_disk_info(const std::vector<dsn::rpc_address> &targets,
+                    /*out*/ std::map<dsn::rpc_address, error_with<query_disk_info_response>> &resps,
```
Comment (acelyc111) [inline include/dsn/dist/replication/replication_ddl_client.h]: If output-only parameters are used, they should appear after input parameters.
-- https://google.github.io/styleguide/cppguide.html#Output_Parameters

**Pull request merged**